source/common/grpc/google_grpc_creds_impl.h

```diff
@@ -13,5 +13,13 @@ grpc::SslCredentialsOptions buildSslOptionsFromConfig(
 std::shared_ptr<grpc::ChannelCredentials>
 getGoogleGrpcChannelCredentials(const envoy::api::v2::core::GrpcService& grpc_service);
 
+std::shared_ptr<grpc::ChannelCredentials>
+defaultSslChannelCredentials(const envoy::api::v2::core::GrpcService& grpc_service_config,
+                             bool allow_insecure = true);
```

Comment:
I don't feel `allow_insecure = true` is right, make it `false` or no default?

Reply:
This PR (other than the new creds stuff) is actually mostly just code movement from what was there already. I agree with your comments and will make said changes.

Reply:
originally, this was set to true to remain consistent with the way it was implemented previously. I agree that call credentials shouldn't be enabled with an insecure channel though, so it makes sense to change it.